src/datamodel/builder/Payload.cpp

```diff
@@ -10,3 +10,100 @@
 // SPDX-License-Identifier: Apache-2.0
 
 #include "up-cpp/datamodel/builder/Payload.h"
+namespace uprotocol::datamodel::builder {
+
+// Byte array constructor
+Payload::Payload(const std::vector<uint8_t>& value_bytes,
+                 v1::UPayloadFormat format) {
+	if (format < v1::UPayloadFormat::UPAYLOAD_FORMAT_UNSPECIFIED ||
+	    format > v1::UPayloadFormat::UPAYLOAD_FORMAT_SHM) {
+		throw std::out_of_range("Invalid byte vector payload format");
+	}
+	std::string serialized(value_bytes.begin(), value_bytes.end());
+	payload_ = std::make_tuple(serialized, format);
+}
+
+// String constructor
+Payload::Payload(const std::string& value, v1::UPayloadFormat format) {
+	if (format < v1::UPayloadFormat::UPAYLOAD_FORMAT_UNSPECIFIED ||
+	    format > v1::UPayloadFormat::UPAYLOAD_FORMAT_SHM) {
+		throw std::out_of_range("Invalid String payload format");
+	}
+	payload_ = std::make_tuple(value, format);
```

Comment:
If there is no need to iterate over payload_, is there a reason for it to be a tuple versus a std::pair, or a struct?
structs are more readable, and aggregate initialization reads better than make_tuples, I think.

Reply:
I would let @gregmedd  comment on this as having tuple was the part of initial design of this class. I am ok to use struct or std::pair

Reply:
The intent is to support using `Payload` only as a temporary wrapper for some information [supporting move semantics and decomposition](https://godbolt.org/#z:OYLghAFBqd5QCxAYwPYBMCmBRdBLAF1QCcAaPECAMzwBtMA7AQwFtMQByARg9KtQYEAysib0QXACx8BBAKoBnTAAUAHpwAMvAFYTStJg1DIApACYAQuYukl9ZATwDKjdAGFUtAK4sGEgKykrgAyeAyYAHI%2BAEaYxCBmABykAA6oCoRODB7evgGp6ZkCoeFRLLHxSbaY9o4CQgRMxAQ5Pn5cgXaYDlkNTQQlkTFxCckKjc2teR22EwNhQ%2BUjSQCUtqhexMjsHOYAzGHI3lgA1CZ7bgReKfTn2CYaAIL7h8eYZxfjxGHAdw/PTxeVCwVBOABEAPIAfQAso8AJIRf5oBjjTCqFLEE7RVCeE4EYhMZAAayhXx%2BCg%2BYPxxC8mHOVkBZj2NSUyIEaIxWJxeIJRNJ5KMlPO1KoYjZe0Zz2ZrjwVH%2B/0YPhORyYCkpADESCwmAQzgB2KUnY0nABSQghEVI/xNJ2UACUIQAVCEWOQa61PW0O52u91QgDq9seymU2DBUMRUMeEQAmp7HrbgwGFfqwQyFU8AG6oPDoGn8qGECAo8YncboEAgQW/MwANhWGcBTy%2BXgcdqYAE9aKgmPmTIabSblF2e32IBWqzXzHWZ%2BXSCctcQdXqqCsTiATilR730FCJwRKyAdcTMFCrjdMAejyxUFmrwo1ic1%2BuB0bbSa%2BSSiwRr1XgJgBDnG4Gh3BA27druUIrI2kpDsaA7ps2iYmqWBDopiKocnqk4gBetyfIeU4Ej8C5Liufz1icAG/uuaFnFRb7wR%2BxCAZsDBbjufZQk2KEIWmmZ8VhqLoVy5ZEXh1wEW4uE1mR2q6nc1GARAr71nOTFeh%2BxqsVcxAcbht73uBXF7rB778UhAKPJieBZrq7DMbh%2BH0oRR5yYuClAXs2CcZB3G8YhgVPCO/n5pimDbqxJYIE0JyqK%2Bg5acaskkUYfljugvG2hBmUAHSsRkABeV5cOZzEAPQVScFidtu6onEIFrPiQJxYLezG5buBWYEoxDGXWkjlclGW7iY/gWKB/jUiK8XZZ%2BhLfsWXV9sNQm6exBoWOJN53leK3oE%2B5G6lWyYGumcGAgJyGpd86Xol%2Bv6hZlM5zikiUWScTBeEQZwTQdC78MuinTVSO1VkZ%2B0rHlNGqfNxpfqSy2mWttpyic1BeVSs3HQQp2PAGH3MbaG36aNfbw%2BdzGkxx5hmHTgXXdZVUs6zbPsxznNc2zCrMq8XinMB94OCQfwthJwtEMQwH0TmeYAFRKV8UI9qIdSorxcv5ojP4lth4PVmltZ1uWRMjejaEYViEA6zWj5bcTJrK6rupZAoeUpF4CgIFC0T8hOeXoLqTCqajJpBVdVm8wcwKYKCkKwgiSJMgcDBHAL7zAU4XyYKwYuPGEeo6mEqkOyN32/SkYMRVFV5gBwyD12HxoV6g5Zgw9i2/oZe3gTBlO4WgP0fG4wHt8BY%2B4a4tCU0DGOt3OtAbuJxAq6gatu2bQm2oPGx6hPFwnEvB9uAb0%2BUxHzxM/sspUBway0Jw/i8H43C8KgnCj5Y1jlhsWyZ8yHgpA8YcC0DBUgxIQD%2BA0PoTgkgX6aHfpwXgCgQAwJAWA0gcBYAwEQCgVALAUh0DiOQSgaBCHEPiMAKQXA%2BB0HQsQVBEBoiININEMITROycCAew5gxBOwQmiNoboICgHkLYIICEDBaBcNAbwLA0QvDADcGIWgqC36kHaoYYA4g5GaLwKxHo951FaCCKoboP0dhAMLjUVhtA8DREJPwjwWBWEkRYNw3g95iA4iUGCTAOojD2KMIgtYVADDAAUAANTwJgAA7hCFIjBPEyEECIMQ7ApCpPkEoNQrDdC0IMCE0w39LD6AcagyAaxUApHVuogAtBCYAJx6n1IrCKEpVhLBcH1O/bx3wsCVNUtUWoWQXAMHcJ4NoegQgLDKBUPQaQMjqymO0AoyysiDHmSMWhXQej1DmKsvQez1Z9GaFs4Y8RdmHKmdMWY/QLlLCuWsBQf9tgSAfk/BBeiP4cHiokOs9TBrUWQMgE4Ug8pcAxrgQgrV9hlV4Bg%2B%2BWCkDkKIfQMgFA9YUIxSAahkhaE0FoAwphLC9G8M4Skil/DBHCIcCk8RjACBSJkawhRSiVG0DUSkrRQSdimPwIYxwxjWHogsehVhNjH56PsY4zhLj%2BWIu%2BB4jR3jfGYH8YEnRPxQl8AidE2JCSkmvyAfwNJohxBZLNTklQ6g9G6DMPobRKBrDWHKdEIZ1Talu04I05prT2l7DBJ06wPSWlggTnCREfS4gDN6vAF5NQRFjIgK4I5tDZmlEuYswoKzblrKWUUBgjyFm7KTfshgZyWj5uOeW05cwS07PuZMGt1yHlzOzWVdYmx3ldulc/Ugr9TG/P%2BYC4FwBQXgskJC6F%2BApYMT2Ai4BoS1iQOgbAjg8DSAeP8IkPKewD2HqPUeusg7WG/JQWg5dci1jYLwXvT2BBSHYvRXECIrAdiqABUCyQIKwUQqAZgOdJA8x6Gteky10hrWKFtfkvQcTCQpE8Z8jgA6h1II4BCH6j6TioFBF%2Bsdv6J3/unVCiAHgcVxAXUupFt6kAPp%2Bs%2BtFlD31sE4ARn9f6p2Qt4EB2FAywOyAg5kqDsgYN5PtQkUgCGmBIbfihtD57OBYYIDhvDo7OPEe42Rijr6sT7DMCsRFK7SAIFzlgeIwzpVbo8QSvKdYHOOac45/UZ6fnINsHoWjq6oEwOlXsb5w6PPeY3WYQLGHjM3rWN4jIzhJBAA%3D%3D).

There is not much point in switching to `std::pair` as that is effectively just a specialized tuple that allows access patterns we don't want to support here.

While a struct is fine, that adds layers that aren't really needed here. The intended use is similar to the example linked above - data (and a format) will be put into a Payload, which will then be moved through the layers of up-cpp to the point where it is needed to populate the UMessage object, at which time the contents will be unpacked and moved into the message.